### ISSUE TITLE
Add Image infrastructure

### ISSUE THREAD

**Issue by mad-moo** [rio-labs/rio#127]
Rio already comes with infrastructure to handle assets. They're hosted for as long as they're needed, and released once all references have been dropped. This infrastracture also currently handles images.

This works, but there is some good reasons to handle images separately:

- Different devices need different images. Based on screen size, it might be reasonable to send lower-resolution images rather than the full blown image
- Images could be converted to more efficient file formats (e.g. png -> jpeg). This comes with its own set of quality related dangers of course.
- The image component on the JS side would really like to know the aspect ratio of images before they're loaded, This is needed e.g. to set the corner radius of the image. If the server were to read each image once (or at least the metadata) it could be sent to the client along the actual image data

Long story short, this would make Rio faster, and also save users from having to know which formats are suited for what type of content.

**Comment by Sn3llius** [comment]
I like the idea! We should also include **Better Image Sizing** (#65)  in our discussion, as we'll need to address the aspect ratio anyway.
